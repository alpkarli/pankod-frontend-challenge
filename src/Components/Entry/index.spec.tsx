import { create } from "react-test-renderer";
import { Entry, MainEntry, EntryPageContent } from "@Components";

const entryProps = {
	title: "Movie1",
	description: "Lorem Ipsum",
	programType: "Movie",
	releaseYear: 2011,
	images: {
		"Poster Art": {
			url: "abc.com/image.jpeg",
		},
	},
};

describe("Entry Components", () => {
	describe("Entry", () => {
		it("should match snapshot", () => {
			const component = create(<Entry {...entryProps} />);
			expect(component.toJSON()).toMatchSnapshot();
			/*  const entryInstance = entry.getInstance();
            expect(entryInstance).toEqual('Movie1'); */
    });
    it("should have the props", () => {

    })
	});
	describe("MainEntry", () => {
		it("should match snapshot", () => {
			const component = create(<MainEntry {...entryProps} />);
			expect(component.toJSON()).toMatchSnapshot();
		});
	});
	describe("EntryPageContent", () => {
		it("should match snapshot", () => {
			const component = create(<EntryPageContent {...entryProps} />);
			expect(component.toJSON()).toMatchSnapshot();
		});
	});
});
