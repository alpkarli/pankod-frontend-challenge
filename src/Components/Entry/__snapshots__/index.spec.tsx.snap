// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Components Entry should match snapshot 1`] = `
<div
  className="entry__container"
>
  <div
    className="entry__content"
  >
    <img
      className="entry__image"
      src="abc.com/image.jpeg"
    />
  </div>
  <p
    className="entry__container__text"
  >
    Movie1
  </p>
</div>
`;

exports[`Entry Components EntryPageContent should match snapshot 1`] = `
<div
  className="entry-page__container"
>
  <h1>
    Movie1
  </h1>
  <div>
    <img
      className="entry-page__image"
      src="abc.com/image.jpeg"
    />
  </div>
  <p>
    <b>
      2011
    </b>
  </p>
  <p>
    Lorem Ipsum
  </p>
</div>
`;

exports[`Entry Components MainEntry should match snapshot 1`] = `
<div
  className="entry__container"
>
  <div
    className="entry__content"
  >
    <h2
      className="entry__content__element"
    >
      MOVIE1
    </h2>
    <img
      className="entry__content__element--image entry__content__element "
      src="abc.com/image.jpeg"
    />
  </div>
  <p
    className="entry__container__text"
  >
    Popular 
    Movie1
  </p>
</div>
`;
